src/util/simplify_expr_int.cpp

```diff
@@ -955,8 +955,13 @@ bool simplify_exprt::simplify_concatenation(exprt &expr)
   // { x } = x
   if(expr.operands().size()==1)
   {
-    exprt tmp;
-    tmp.swap(expr.op0());
+    exprt tmp = expr.op0();
+    if(expr.type() != tmp.type())
+    {
+      tmp.make_typecast(expr.type());
+      simplify_node(tmp);
+    }
```

Comment:
This one is potentially problematic, as the typecast is semantic, and does not necessarily preserve the bit-pattern.
I would refuse to do the simplification if the type differs.

Reply:
Ok, I’ll change this as suggested.